drivers/gles2/rasterizer_canvas_gles2.h

```diff
@@ -58,8 +59,71 @@ class RasterizerCanvasGLES2 : public RasterizerCanvas {
 		float time;
 	};
 
-	struct Data {
+	// pod versions of vector and color and RID, need to be 32 bit for vertex format
+	struct BatchVector2 {
+		float x, y;
+		void set(const Vector2 &o) {
+			x = o.x;
+			y = o.y;
+		}
+		void to(Vector2 &o) const {
+			o.x = x;
+			o.y = y;
+		}
+	};
+
+	struct BatchColor {
+		float r, g, b, a;
+		void set(const Color &c) {
+			r = c.r;
+			g = c.g;
+			b = c.b;
+			a = c.a;
+		}
+		bool equals(const Color &c) const {
+			return (r == c.r) && (g == c.g) && (b == c.b) && (a == c.a);
+		}
+		const float *get_data() const { return &r; }
+	};
+
+	struct BatchVert {
+		// must be 32 bit pod
+		BatchVector2 pos;
+		BatchVector2 uv;
+	};
+
+	struct BatchVert_Colored : public BatchVert {
+		// must be 32 bit pod
+		BatchColor col;
+	};
+
+	struct Batch {
+		enum eType : uint32_t {
+			BT_DEFAULT,
+			BT_RECT,
+		};
+
+		eType type;
+		uint32_t first_command;
+		uint32_t num_commands;
+		uint32_t first_quad;
+		uint32_t btex_id;
```

Comment:
should this be "batch_texture_id"? If the "b" doesnt stand for batch, then should be "texture_id"